addon/adapters/twitch.js

```diff
@@ -1,35 +1,47 @@
 import JSONAPIAdapter from 'ember-data/adapters/json-api';
 import Inflector from 'ember-inflector';
-import { camelize } from 'ember-string';
+import TwitchConfig from 'ember-data-twitch/configuration';
 
 const { inflector } = Inflector;
 
-
 export default JSONAPIAdapter.extend({
   host: 'https://api.twitch.tv',
   namespace: 'kraken',
-  dataType: 'jsonp',
+  defaultSerializer: '-twitch',
   pluralizePath: true,
 
+  init() {
+    this._super(...arguments);
+
+    if (!TwitchConfig.clientID) {
+      throw new Error(`please set a "clientID" in the ENV['ember-data-twitch'] hash of \`config/environment.js\``);
+    }
+  },
+
+  headers: {
+    Accept: 'application/vnd.twitchtv.v3+json',
+    'Client-ID': TwitchConfig.clientID
```

Comment:
Is there ever a reason we would want to set `clientID` on the adapter and allow it to be overridden by other adapters that extend the base adapter? If not, this seems ok.